bentoml/_internal/frameworks/transformers.py

```diff
@@ -106,427 +65,209 @@ def check_flax_supported() -> None:  # pragma: no cover
-            "model_kwargs", {}
-        )  # type: t.Dict[str, t.Any]
-        self._kwargs = pipeline_kwargs
-
-        # tokenizer-related
-        try:
-            self._has_tokenizer = (
-                self.model_info.info.options["feature_extractor"] is False
-            )
-        except Exception:
-            self._has_tokenizer = False
-        self._tokenizer = None
-        self._pipeline = None
-        self._config = None
-
-    @property
-    def num_replica(self) -> int:
-        # TODO: supports multiple GPUS
-        return 1
-
-    def _setup(self) -> None:
-        params = load(
-            self._tag,
-            from_flax="flax" in self._framework,
-            from_tf="tf" in self._framework,
-            return_config=True,
-            revision=self._revision,
-            use_fast=self._use_fast,
-            use_auth_token=self._use_auth_token,
-            model_store=self.model_store,
-            **self._kwargs,
-        )
+    class TransformersRunnable(bentoml.Runnable):
+        SUPPORT_NVIDIA_GPU = True  # type: ignore
+        SUPPORT_CPU_MULTI_THREADING = True  # type: ignore
+
+        def __init__(self):
+            super().__init__()
+            task = bento_model.info.options["task"]
+            try:
+                transformers.pipelines.check_task(task)  # type: ignore
+            except KeyError as e:
+                raise BentoMLException(
+                    f"{e}, as `{task}` is not recognized by transformers."
+                )
 
-        if self.model_info.info.context["pipeline"]:
-            self._pipeline = params
-        else:
-            self._config, self._model, _tfe = params
-            if not self._has_tokenizer:
-                self._feature_extractor = _tfe
+            available_gpus = os.getenv("NVIDIA_VISIBLE_DEVICES")
+            if available_gpus is not None and available_gpus != "":
+                # assign GPU resources
+                kwargs = {
+                    "device": available_gpus,
+                }
             else:
-                self._tokenizer = _tfe
-            self._pipeline = transformers.pipeline(
-                self._tasks,
-                config=self._config,
-                model=self._model,
-                tokenizer=self._tokenizer,  # type: ignore[reportGeneralTypeIssues]
-                framework=self._framework,
-                feature_extractor=self._feature_extractor,
-                revision=self._revision,
-                use_fast=self._use_fast,
-                use_auth_token=self._use_auth_token,
-                model_kwargs=self._model_kwargs,
-                device=self._device,
-                **self._kwargs,
-            )
-
-    def _run_batch(self, *args: t.Any, **kwargs: t.Any) -> t.Any:
-        return self._pipeline(*args, **kwargs)  # type: ignore
+                # assign CPU resources
+                kwargs = {}
 
+            self.pipeline = load_model(bento_model, **kwargs)
```

Comment:
this kwargs would only have one option `device`, so if users want to pass in other transformers kwargs this would not work. I think we should make a 1-to-1 map from transformers kwargs to our load kwargs and save it under `ModelOptions`

Reply:
or shall we put them under with_args?

Reply:
How do we get an exhaustive list of Transformers pipeline argument? I was also not clear what you meant by saving it under `ModelOptions` under the `load_model()` path.

Reply:
I think it is impossible to get an exhaustive list of all of the pipeline arguments. I think one way is that we accept `**kwargs` for `get_runnable`, and then append our default to that kwargs if user is not set. Then that `kwargs` can be passed to `load_model`